### PR TITLE
CIN-03643: save fails when removing and then adding images on a link-multichoice field

### PR DESCRIPTION
**Pull request by DAquilinaCinchy**
Corrected incorrect parsing logic when saving the field's value and added null checks in the logic for saving the whole form.

This version targets the latest version of Cinchy's Forms app

### CODE REVIEW

**Review by alvin-h — APPROVED** [review]

File: src/app/core/sidenav/sidenav.component.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class SidenavComponent implements OnInit, OnDestroy {
 
     this.appStateService.onRecordSelected().subscribe(resp => {
       this.rowId = resp.cinchyId;
-      this.filteredTableUrl = !this.rowId.toString().includes('null') ? `${this.tableUrl}?viewId=0&fil[Cinchy%20Id].Op=Equals&fil[Cinchy%20Id].Val=${this.rowId}` : this.tableUrl;
```
Comment (alvin-h) [inline src/app/core/sidenav/sidenav.component.ts:59]: Our forms ids can be a string, it's driven off the [Form ID] column. Row Id can also be the literal string "null".

Eg: <forms-url>?formId=myform&rowId=null

Will this be an issue?

Comment (cinchy-m) [reply src/app/core/sidenav/sidenav.component.ts:59]: `?formId=myform&rowId=nullzzzzzzz` (anything other than `null` that isn't a valid/existing id) throws entitlements errors. Omitting `&rowId` altogether doesn't throw errors but will display the last loaded record - e.g. if you previously loaded `&rowId=3` - which is odd.

If you consider this to be in scope of these changes, ideally we'd show no form and a message communicating no record with that ID exists but that needs a bit more consideration so silently failing (don't show errors) and just showing the blank/new form would be sufficient.

Comment (alvin-h) [reply src/app/core/sidenav/sidenav.component.ts:59]: ignore the form id one, I misread the change. I guess you made it always string instead, that makes sense to me. But for the 'null' checks on rowId, would it cause an issue?

Comment (DAquilinaCinchy) [reply src/app/core/sidenav/sidenav.component.ts:59]: "null" is no longer a possible value for the rowId. I've added logic to explicitly eliminate that case. I'll add in some more graceful handling for mangled rowIds.

Comment (DAquilinaCinchy) [reply src/app/core/sidenav/sidenav.component.ts:59]: I'll also add in some better handling for cases where the session information conflicts with the current form for missing query params. In particular the case where the rowId isn't included, and the value stored in the session isn't valid for the current form.

**Review by alvin-h — APPROVED** [review]

**Review by cinchy-m — DISMISSED** [review]

**Review by alvin-h — DISMISSED** [review]

**Pull request merged**